client/src/views/Signup.vue

```diff
@@ -0,0 +1,199 @@
+                             </label>
+                             <input 
+                                 v-model="email"
+                                 type="email" 
+                                 name="email" 
+                                 id="email" 
+                                 class="bg-transparent border-gray-600 text-gray-900 sm:text-lg rounded-lg focus:ring-primary-600
+                                         focus:border-primary-600 block w-full p-3" 
+                                 placeholder="name@company.com" 
+                                 required
+                             >
+                         </div>
+                         <div>
+                             <label 
+                                 for="password" 
+                                 class="block mb-2 text-lg font-medium text-gray-900"
+                             >
+                                 Password
+                             </label>
+                             <input 
+                                 v-model="password"
+                                 type="password" 
+                                 name="password" 
+                                 id="password" 
+                                 placeholder="••••••••" 
+                                 class="bg-transparent border-gray-600 text-gray-900 sm:text-lg rounded-lg focus:ring-primary-600 
+                                          focus:border-primary-600 block w-full p-3" 
+                                 required
+                             >
+                         </div>
+                         <BaseButton type="submit" class="w-full" button-title="Sign up" color="black"></BaseButton>
+                         
+                         <p class="text-lg text-gray-500">
+                             Already have an account? <a href="/login" class="font-semibold hover:underline">Log in</a>
+                         </p>
+
+                         <p v-if="isShowingError" class="font-primary font-semibold text-lg text-red-600">User with this email already exists</p>
+                     </form>
+                 </div>
+                 
+             </div>
+
+             <div class="lg:flex lg:w-[30rem] py-10 hidden">
+                <div class="bg-[url('../assets/img/background.png')] bg-cover bg-center bg-no-repeat w-full rounded-r-xl drop">
+ 
+                </div>
+             </div>
+             </div>
+         </div>
+     </div>
+ </template>
+ 
+<script setup lang="ts">
+
+import { ref } from 'vue';
+
+import BaseButton from '../components/base-components/BaseButton.vue';
+import todoService from '../service/todoService.ts';
+import { useRouter } from 'vue-router';
+
+const firstName = ref('');
+const lastName = ref('');
+const email = ref('');
+const password = ref('');
+
+const router = useRouter(); 
+const userId = ref('');
+const isShowingError = ref(false)
+
+const createUser = async () => {
```

Comment:
We discussed that methods in vue components should look like this:

async function createUser() {